http://sourceware.org/ml/gdb-patches/2012-07/msg00200.html
Subject: [obv] testsuite: Fix fission-reread.S regression by me

Hi Doug,

I broke the testcase by patch
	[patch 1/2] testcase: Make it -m32 compatible [Re: [commit] Fix rereading of DWO DIEs]
	http://sourceware.org/ml/gdb-patches/2012-07/msg00009.html

The testcase itself usually PASSes but sometimes it does not.  It is more
clear with 'readelf -wi':
    <7b>   DW_AT_frame_base  : 97 byte block: 7a 0 1 7 19 8b 7e ac ce f1 22 90 2 fb 1 0 0 0 0 0 0 0 0 0 0 0 0 0 51 cd 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 	(DW_OP_breg10 (r10): 0; (Unknown location op))

Checked in.


Sorry,
Jan


http://sourceware.org/ml/gdb-cvs/2012-07/msg00102.html

--- src/gdb/testsuite/ChangeLog	2012/07/13 20:26:10	1.3292
+++ src/gdb/testsuite/ChangeLog	2012/07/15 08:34:55	1.3293
@@ -1,3 +1,9 @@
+2012-07-15  Jan Kratochvil  <jan.kratochvil@redhat.com>
+
+	Fix a testcase regression by me.
+	* gdb.dwarf2/fission-reread.S: Fix two DW_FORM_data8 to be
+	DW_FORM_data4.
+
 2012-07-13  Jan Kratochvil  <jan.kratochvil@redhat.com>
 	    Doug Evans  <dje@google.com>
 
--- src/gdb/testsuite/gdb.dwarf2/fission-reread.S	2012/07/02 20:10:48	1.3
+++ src/gdb/testsuite/gdb.dwarf2/fission-reread.S	2012/07/15 08:34:55	1.4
@@ -231,7 +231,7 @@
 	.uleb128 0x11	/* (DW_AT_low_pc) */
 	.uleb128 0x1	/* (DW_FORM_addr) */
 	.uleb128 0x12	/* (DW_AT_high_pc) */
-	.uleb128 0x7	/* (DW_FORM_data8) */
+	.uleb128 0x6	/* (DW_FORM_data4) */
 	.uleb128 0x10	/* (DW_AT_stmt_list) */
 	.uleb128 0x17	/* (DW_FORM_sec_offset) */
 	.uleb128 0x1b	/* (DW_AT_comp_dir) */
@@ -351,7 +351,7 @@
 	.uleb128 0x11	/* (DW_AT_low_pc) */
 	.uleb128 0x1f01	/* (DW_FORM_GNU_addr_index) */
 	.uleb128 0x12	/* (DW_AT_high_pc) */
-	.uleb128 0x7	/* (DW_FORM_data8) */
+	.uleb128 0x6	/* (DW_FORM_data4) */
 	.uleb128 0x40	/* (DW_AT_frame_base) */
 	.uleb128 0x18	/* (DW_FORM_exprloc) */
 	.byte	0

